src/_tests/integration/integration-utils.js

```diff
@@ -1,5 +1,38 @@
+const waitForFrame = async url => {
+  const frameLoaded = new Promise(resolve => {
+    const navHandler = async frame => {
+      await frame.url().includes(url) && resolve({ frame, navHandler })
+    }
+    page.on('framenavigated', navHandler)
```

Comment:
TOL: Is there any risk that the frame will already be navigated by the time we call this?

Reply:
Yes, that is a concern of mine as well. I made a slight tweak to the URL in DataExplorerFrame to conform to what page.frames() returns. This is currently working